src/server/Adaptive.ReactiveTrader.Server.Pricing/PriceSource.cs

```diff
@@ -84,21 +79,48 @@ private static IPriceGenerator CreatePriceGenerator(string symbol, decimal initi
       return new MeanReversionRandomWalkPriceGenerator(symbol, initial, precision);
     }
 
-    private IObservable<Unit> RegisterPriceTrigger(string symbol)
+    private void RefreshMarketRates(DateTime refreshDateTime)
+    {
+      foreach (var adapter in _marketAdapters)
+      {
+        var marketData = new List<MarketData>(adapter.GetMarketData().Result);
```

Comment:
We shouldn't be using the Result of a task directly

Reply:
Also, we need to handle the case if a market data source is down/the adapter fails for whatever reason.
We should probably fall back to the hard coded values in that case

Reply:
1. Done.
2. Agreed. The PriceSource class initializes itself with all the required currency pairs, with the same hard-coded defaults as before. Then every hour, the 3 adapters are called (primary one first). This means if the EuropeanCentralBank, FinancialModellingPrep, and Yahoo are all down then the defaults are used.